dags/elsevier/repository.py

```diff
@@ -0,0 +1,61 @@
+import os
+from io import BytesIO
+from typing import IO
+
+from common.repository import IRepository
+from common.s3_service import S3Service
+from common.utils import find_extension
+
+
+class ElsevierRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.s3 = S3Service(os.getenv("ELSEVIER_BUCKET_NAME", "elsevier"))
+
+    def get_all_raw_filenames(self):
+        return [
+            f.key.replace("raw/", "")
+            for f in self.s3.objects.filter(Prefix=self.ZIPED_DIR).all()
+        ]
+
+    def find_all(self, filenames_to_process=None):
+        grouped_files = {}
+        filenames = (
+            filenames_to_process
+            if filenames_to_process
+            else self.__find_all_extracted_files()
+        )
+        for file in filenames:
+            last_part = os.path.basename(file)
+            filename_without_extension = last_part.split(".")[0]
+            if filename_without_extension not in grouped_files:
+                grouped_files[filename_without_extension] = {}
+            extension = find_extension(last_part)
+            grouped_files[filename_without_extension][extension] = file
```

Comment:
I don't understand why you need the grouping, the whole function looks a bit overcomplicated to me

Reply:
the idea of this function:
We can have files that have the same name but different extensions. 

A.xml.zip
A.pdf.zip
A.archival.zip

B.xml.zip
B.pdf.zip
B.archival.zip

So I take the name (A) and group files by it:
[
  {
            "pdf": ""extracted/date_A.pdf",
            "xml": ""extracted/date_A.xml",
            "pdfa": "extracted/date_A.pdfa"
        },
  {
            "pdf": ""extracted/date_B.pdf",
            "xml": ""extracted/date_B.xml",
            "pdfa": "extracted/date_B.pdfa"
        }
]